File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMApp/Core/Analytics/CompositeAnalyticsService.swift
import Foundation

/// Fan-out аналитика: делегирует события всем подключённым сервисам.
final class CompositeAnalyticsService: AnalyticsService {

    private let services: [AnalyticsService]

    init(services: [AnalyticsService]) {
        self.services = services
    }

    func track(_ event: AnalyticsEvent) {
        services.forEach { $0.track(event) }
    }

    func setUser(userId: Int, role: String) {
        services.forEach { $0.setUser(userId: userId, role: role) }
    }

    func resetUser() {
        services.forEach { $0.resetUser() }
    }
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/android/app/src/main/java/com/beyondviolet/wfm/core/analytics/CompositeAnalyticsService.kt
package com.beyondviolet.wfm.core.analytics

/**
 * Fan-out аналитика: делегирует события всем подключённым сервисам.
 */
class CompositeAnalyticsService(private val services: List<AnalyticsService>) : AnalyticsService {

    override fun track(event: AnalyticsEvent) = services.forEach { it.track(event) }

    override fun setUser(userId: Int, role: String) = services.forEach { it.setUser(userId, role) }

    override fun resetUser() = services.forEach { it.resetUser() }
}
